MarkBind/markbind
# Remove duplicate code in syntax doc

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Documentation update
• [ ] Bug fix
• [ ] New feature
• [ ] Enhancement to an existing feature
• [ ] Other, please explain:



Fixes #1159
~~Requires #1193~~ (merged) 



**What is the rationale for this request?**

Currently developers need to write the same code twice for syntax and component documentation: one for the code block in the "code" section and one to be rendered by MarkBind in the "output" section. However this duplication should have been avoided. And developers may forget to make sure that both sections are on parity with each other.

**What changes did you make? (Give an overview)**

Created a new boilerplate `codeAndOuput.md` that takes in MarkBind code assigned to the `code` variable declared in the main markdown file.
And renders two things: (1) the code block containing the code, and (2) the rendered output of the code

**Provide some example code that this change will affect:**

**Is there anything you'd like reviewers to focus on?**


**Testing instructions:**

Compare the following pages, and see if any of the examples regress or render wrongly, be it either the source code or the rendered output.

https://deploy-preview-1168--markbind-master.netlify.com/userguide/formattingcontents
https://markbind.org/userGuide/formattingContents.html

https://deploy-preview-1168--markbind-master.netlify.com/userguide/usingcomponents
https://markbind.org/userGuide/usingComponents.html


**Proposed commit message: (wrap lines at 72 characters)**
Remove duplicate code in syntax documentation

Right now, we have to type example code in our User Guides twice,
once as example syntax, and once to be rendered by Markbind.

This practice is prone to errors as we have to ensure the two code
are the same. And it makes the user documentation quite lengthy
in terms of the source md and mbdf files. 

Let's make a boilerplate that takes the example code, 
and renders it once as a code block containing the code
and once as the rendered MarkBind output of the code.
And use it in the User Guide.
